clang/lib/CConv/RewriteUtils.cpp

```diff
@@ -1247,42 +1248,44 @@ class CastPlacementVisitor :
     if (O->getOpcode() == BO_Assign) {
       Expr *LHS = O->getLHS();
       Expr *RHS = O->getRHS();
-      auto &CS = Info.getConstraints();
       std::set<ConstraintVariable *> LCons = CR.getExprConstraintVars(LHS);
-      ConstraintVariable *LCVariable = nullptr;
-      bool LHSChkType = false;
-      for (auto *LC : LCons) {
-        if (LC->anyChanges(CS.getVariables())) {
-          LCVariable = LC;
-          LHSChkType = true;
+      std::string CastStr;
+      if (needsFancyCast(LCons, RHS, CastStr)) {
+        surroundByCast(CastStr, RHS);
+      }
+    }
+    return true;
+  }
+
+  bool VisitReturnStmt(ReturnStmt *S) {
+    Expr *RetExpr = S->getRetValue();
+
+    if (RetExpr != nullptr) {
+      assert(FD != nullptr && "Function Declaration should not be nullptr");
+
+      std::set<ConstraintVariable *> Fun = Info.getVariable(FD, Context);
+
+      std::set<ConstraintVariable *> FuncRet;
+      for (const auto &F : Fun) {
+        if (FVConstraint *FV = dyn_cast<FVConstraint>(F)) {
+          FuncRet = FV->getReturnVars();
           break;
```

Comment:
Is this an assumption that `Fun` will have only one thing in it? Use `Utils::getOnly` then?